geonode/geoserver/helpers.py

```diff
@@ -2280,6 +2280,22 @@ def get_dataset_type(spatial_files):
     return the_dataset_type
 
 
+def get_dataset_capabilities_url(layer, version="1.3.0", access_token=None):
+    """
+    Generate the layer-specific GetCapabilities URL
+    """
+    workspace, layername = layer.alternate.split(":") if ":" in layer.alternate else (None, layer.alternate)
+    wms_url = ogc_server_settings.LOCATION
+    if not layer.remote_service:
+        wms_url = f"{wms_url}{workspace}/{layername}/wms?service=wms&version={version}&request=GetCapabilities"  # noqa
```

Comment:
I'm afraid here that if `workspace == None` the statement `{workspace}` will translate it wrongly. Maybe bettere to put `('', layer.alternate)` above

Reply:
done